### DISCUSSION TITLE
FlexibleColumnLayout + AnalyticalTable calculation of height

### DISCUSSION THREAD

**Discussion by DrMalchev** [UI5/webcomponents-react#6234] [Q&A]
We have an issue with heights since a long time that we solve with custom heighcalculation functions and JS observers, but I think there should be a better way to do this. The structure in pseudo-code is as follows:

`const [numberOfRows, setNumberOfRows] = React.useState(15)

<FlexibleColumnLayout
   style={{ height: calculateHeight() }}
   startColumn={
      <>
      <slot_>
         <FilterBarComponent /_>
      </slot_>
      <AnalyticalTable
      visibleRows={numberOfRows}
         ............. />
      </>
      }
/>
`
Number of rows is calculated based on the available space, so is the height of the FlexColumnLayout.
I think we may be using the components in a wrong way.
Thanks.

**Comment by Lukas742** [comment]
Hi @DrMalchev 

I don't think that it's possible implementing this scenario without an observer, at least I didn't find one. 
I believe your implementation can still be simplified though, since you don't need to calculate the number of rows yourself, but use the `AnalyticalTableVisibleRowCountMode` `Auto` or `AutoWithEmptyRows`.

I've prepared a StackBlitz example that shows how this could be achieved: https://stackblitz.com/edit/ui5wcr-fcl-fb-table-spread?file=src%2FApp.tsx,src%2Fdata.json
